gallery_dl/extractor/boosty.py

```diff
@@ -219,6 +219,107 @@ def items(self):
             yield Message.Queue, url, user
 
 
+class BoostyDirectMessagesExtractor(BoostyExtractor):
+    """Extractor for boosty.to direct messages"""
+    subcategory = "direct-messages"
+    directory_fmt = ("{category}", "{user[blogUrl]} ({user[id]})",
+                     "direct-messages")
+    pattern = BASE_PATTERN + r"/app/messages\?dialogId=(\d+)"
+    example = "https://boosty.to/app/messages?dialogId=123456"
+
+    def items(self):
+        """Yield direct messages from a given dialog ID."""
+        dialog_id = self.groups[0]
+        response = self.api.dialog(dialog_id)
+        sign = response.get("signedQuery", "")
+        messages = response.get("messages", {}).get("data", [])
+        chatmate = response.get("chatmate", {})
+        user = self.api.user(chatmate.get("url", ""))
+
+        for message in messages:
+            files = self._process_dialog(message, sign=sign)
+
+            if files:
+                data = {
+                    "post": message,
+                    "user": user,
+                    "count": len(files),
+                }
+                yield Message.Directory, data
+
+                for data["num"], file in enumerate(files, 1):
+                    data["file"] = file
+                    url = file["url"]
+                    yield Message.Url, url, text.nameext_from_url(url, data)
+
+        for message in self.api.dialog_messages(dialog_id=dialog_id,
+                                                offset=messages[0]["id"]):
```

Comment:
That's the same code a second time. Use `itertools.chain` to combine the original `messages` with the additional ones:

``` python
messages = itertools.chain(
    messages, self.api.dialog_messages(
        dialog_id, offset=messages[0]["id"]))
for message in messages:
    ...
```

> offset=messages[0]["id"]

Is using the first message's `id` correct here?